torch/csrc/jit/frontend/tracer.cpp

```diff
@@ -459,9 +462,26 @@ std::pair<std::shared_ptr<TracingState>, Stack> trace(
       gatherParametersAndBuffers(state, self_value, *self, {"__module"});
     }
 
-    for (IValue& input : inputs) {
-      input = addInput(state, input, input.type(), state->graph->addInput());
+    // When enough argument name hints are provided, use them as debug names
+    // for traced function/modules.
+    // Here argument_names is allowed to have more names than needed because
+    // some arguments may have valid default values, therefore they don't need
+    // example inputs.
+    if (argument_names.size() >= inputs.size()) {
```

Comment:
What's the justification for not using `argument_names` at all if there aren't enough for all inputs? Not sure how this can happen, but isn't having some meaningful argument names better than having none?

Reply:
Because I can't think of a case where `argument_names.size() < inputs.size()` can happen legally. The only way this can happen is if we special-case another method (similar to how `forward` is special-cased on tracer). Thus leaning on the safer side to back-out from assigning argument name. Let me know what you think.